bridge/discord/discord.go

```diff
@@ -323,6 +326,37 @@ func (b *Bdiscord) getNick(user *discordgo.User) string {
 	return user.Username
 }
 
+func (b *Bdiscord) getGuildMemberByNick(nick string) (*discordgo.Member, error) {
```

Comment:
This function will have locking issues, you can't range over the userMemberMap without locking.
I think it's better to just make a new map called nickMemberMap which has the nick as key. 

So this can be rewritten analogue to the existing `getNick` function.

Also update the `memberUpdate` function to add the new nick to `nickMemberMap`